tooling/lsp/src/notifications/mod.rs

```diff
@@ -42,7 +43,38 @@ pub(super) fn on_did_change_text_document(
     params: DidChangeTextDocumentParams,
 ) -> ControlFlow<Result<(), async_lsp::Error>> {
     let text = params.content_changes.into_iter().next().unwrap().text;
-    state.input_files.insert(params.text_document.uri.to_string(), text);
+    state.input_files.insert(params.text_document.uri.to_string(), text.clone());
+
+    let (mut context, crate_id) = nargo::prepare_source(text);
+    let _ = check_crate(&mut context, crate_id, false, false);
+
+    let workspace = match resolve_workspace_for_source_path(
```

Comment:
similar comment here, to do some unwrap_or_err something

Reply:
We can't do this here as easily as we're returning a `ControlFlow` so we can't just use the `?` syntax unfortunately.